website/docs/commands/publishing/publish-setup.md

```diff
@@ -0,0 +1,250 @@
+---
+title: Publish Setup
+sidebar_position: 19
+---
+
+import {ChainedSnippets} from "../../../src/components/MarkdownComponents.js";
+
+The `publish setup` sub-command configures your project for publishing to Maven repositories,
+such as Maven Central or GitHub Packages. It checks that all required parameters for publishing are set, and tries
+to infer many of them from the environment. It writes configuration as using directives, appended
+to a file named `publish-conf.scala` at the root of the workspace.
+
+`publish setup` can configure publishing, so that you can publish from your local machine,
+but also from GitHub actions.
+In particular, it can upload secrets as GitHub repository secrets, so that only
+minimal effort is required to setup publishing from GitHub actions.
+
+Running `publish setup` works fine on brand new machines or projects, but it works better when:
+- user details, credentials for publishing, and a PGP key are configured (user-wide)
+- the current project is already pushed to GitHub
+
+## Configuration
+
+Configuring Scala CLI for publishing makes `publish setup` works more smoothly later on.
+
+In particular, one can configure:
+- details about oneself (name, email, website - written in the developers section of POM files)
+- a PGP key pair (to sign artifacts)
+- Sonatype credentials (to upload artifacts to Maven Central)
+- a GitHub token (to upload repository secrets to GitHub, and artifacts to Maven repositories of GitHub Packages)
```

Comment:
Shouldn't we also set up github actions with release flow? (snapshot, releases) ?